utf8.c

```diff
@@ -1634,12 +1634,12 @@ Perl_utf8_to_uv_msgs_helper_(const U8 * const s0,
      * than a single character */
     const U8 * send = e;
 
-    SSize_t curlen = send - s0;
+    Size_t curlen = send - s0;
```

Comment:
Why can't this be negative?

Reply:
It can be negative in non-DEBUGGING builds, where `s0 <= send` is treated as a reported error.  For DEBUGGING builds perl asserts.

Reply:
Thank you for comments.  I've forgotten a possibility of `s0 > send`.
I pushed a new commit to move this subtraction to the point after checking `s0 < send` to avoid underflow.